idx/memory/tag_query.go

```diff
@@ -368,9 +367,19 @@ func NewTagQuery(expressions []string, from int64) (TagQuery, error) {
 // getInitialByEqual generates the initial resultset by executing the given equal expression
 func (q *TagQuery) getInitialByEqual(expr kv, idCh chan schema.MKey, stopCh chan struct{}) {
 	defer q.wg.Done()
+	key, err := idx.IdxIntern.GetPtrFromByte([]byte(expr.key))
+	if err != nil {
+		log.Error("memory-idx: Failed to retrieve uintptr for interned tag key: ", err)
+		internError.Inc()
```

Comment:
if `key` does not get set, then the rest of this function and query is not going to result in anything useful, so i think at this point the query might as well be aborted. but i think instead of returning an error from `getInitialByEqual()` and all the other `getInitialBy*` functions it would probably be better to do this lookup before they get called in `getInitialIds()`, so that if there's an error returned from the interning layer the query can be aborted at that point already.

Reply:
Yes, it does require some changes to the method definitions, but it should be better overall I think